### PR TITLE
Update front end to batch responses

### PR DESCRIPTION
**Pull request by kwunkiuma**
The front end now uses the batched reaction handling.

The message once again refreshes automatically on signups.

Managed to avoid adding the client to the buffer with the trade-off of removing DM confirmations, which would have been difficult to implement in batches anyways as invalid responses are silently ignored.

### CODE REVIEW

File: DiscordBot/Program.cs [hunk 1]
```diff
@@ -47,102 +45,75 @@ private static async Task MainAsync()
...
-                await OnMessageReactionAdded(client, eventArguments, eventsSheetsService);
+                await OnMessageReactionAdded(client, eventArguments, eventsSheetsService, reactionBuffer);
             };
 
             await discord.ConnectAsync();
             await Task.Delay(-1);
         }
 
-        private static async Task OnMessageReactionAdded(
+        private static Task OnMessageReactionAdded(
```
Comment (FridaTveit) [inline DiscordBot/Program.cs:60]: Does this need to return a Task at all, or could it just have void return type? 🙂

File: DiscordBot/Program.cs [hunk 2]
```diff
@@ -47,102 +45,75 @@ private static async Task MainAsync()
...
+            // Skip if reaction is not in the signup channel
+            if (eventArguments.Channel.Name != EventCommands.SignupChannelName)
             {
-                return;
+                return Task.CompletedTask;
             }
 
-            await ProcessReaction(client, eventArguments, discordEvent, eventsSheetsService);
+            _ = Task.Run(async () =>
+            {
```
Comment (FridaTveit) [inline DiscordBot/Program.cs:79]: Does this need to be inside a block? 🙂

File: DiscordBot/Program.cs [hunk 3]
```diff
@@ -47,102 +45,75 @@ private static async Task MainAsync()
...
             MessageReactionAddEventArgs eventArguments,
-            DiscordEvent discordEvent,
             IEventsSheetsService eventsSheetsService)
         {
-            var signupsByResponse = await eventsSheetsService.GetSignupsByResponseAsync(discordEvent.Key);
-            await eventArguments.Message.ModifyAsync(
-                eventArguments.Message.Content,
-                GetSignupEmbed(discordEvent, signupsByResponse).Build()
-            );
+
```
Comment (FridaTveit) [inline DiscordBot/Program.cs:107]: Unneeded empty line?

**Review by FridaTveit — COMMENTED** [review]
A few comments, otherwise looks good 🙂

File: DiscordBot/Program.cs (hunk 2 above)
Comment (kwunkiuma) [reply DiscordBot/Program.cs:79]: Nope, removed.

**Review by FridaTveit — APPROVED** [review]
Looks good 🙂

File: DiscordBot/Program.cs (hunk 1 above)
Comment (benjimarshall) [reply DiscordBot/Program.cs]: I've just had a look at this with @kwunkiuma. Basically, the `discord.MessageReactionAdded` event handler on line 51 expects to receive a `Task`, but there's no need for `OnMessageReactionAdded` to returning one in all cases. I've changed it so it returns `void` and the hander here returns a `Task.CompletedTask` in 8c81656 🙂

**Pull request merged**